debian/rules

```diff
@@ -1,33 +1,70 @@
 #!/usr/bin/make -f
+#include /usr/share/dpkg/default.mk
+#include /usr/share/dh-php/pkg-pecl.mk
 
 DH_VERBOSE = 1
 
-DPKG_EXPORT_BUILDFLAGS = 1
-include /usr/share/dpkg/default.mk
+# already in pkg-pecl.mk
+#DPKG_EXPORT_BUILDFLAGS = 1
 
-export DEB_BUILD_MAINT_OPTIONS = hardening=+all
+# already in pkg-pecl.mk
+#export DEB_BUILD_MAINT_OPTIONS = hardening=+all
 
-%:
-	dh $@ 
+# package name should be snuffleupagus, not php-snuffleupagus
+PECL_NAME_OVERRIDE := snuffleupagus
+DH_PHP_VERSIONS_OVERRIDE := $(shell /usr/sbin/phpquery -V)
+include /usr/share/dh-php/pkg-pecl.mk
 
-override_dh_auto_clean:
-	cd ./src; phpize --clean
+# target directory is debian/snuffleupagus, not debian/php-snuffleupagus
+INSTALL_ROOT = $(CURDIR)/debian/$(PECL_NAME)
 
-override_dh_auto_configure:
-	cd ./src; phpize
-	cd ./src; ./configure --enable-snuffleupagus
+DIR_TARGETS = $(addprefix build-,$(DH_PHP_VERSIONS))
+#$(warning CONFIGURE_TARGETS: $(CONFIGURE_TARGETS))
+#$(warning BUILD_TARGETS: $(BUILD_TARGETS))
+#$(warning INSTALL_TARGETS: $(INSTALL_TARGETS))
 
-override_dh_auto_build:
-	make -C src
+PECL_SOURCE=$(filter-out debian $(DIR_TARGETS),$(wildcard *))
+$(foreach ver,$(DH_PHP_VERSIONS),$(eval PECL_SOURCE_$(ver) := $(PECL_SOURCE)))
+$(foreach ver,$(DH_PHP_VERSIONS),$(mkdir build-$(ver)))
 
-override_dh_auto_install:
+override_dh_testdir: $(DIR_TARGETS)
 	dh_testdir
-	dh_testroot
-	dh_installdirs
-	mkdir -p debian/snuffleupagus/$$(/usr/bin/php-config --extension-dir)/
-	mkdir -p debian/snuffleupagus/etc/php/$$(/usr/bin/php-config --version | cut -d '.' -f1-2)/mods-available/
-	install -m 644 -o root -g root src/modules/snuffleupagus.so debian/snuffleupagus/$$(/usr/bin/php-config --extension-dir)/snuffleupagus.so
-	install -m 644 -o root -g root config/snuffleupagus.ini debian/snuffleupagus/etc/php/$$(/usr/bin/php-config --version | cut -d '.' -f1-2)/mods-available/snuffleupagus.ini
-
-override_dh_auto_test:
-	TEST_PHP_ARGS="-q" REPORT_EXIST_STATUS=1 make -C src test
+
+build-%:
+	mkdir -p build-$(*)
+
+# disable 'php-' prefix
+override_dh_php:
+	dh_php -p $(PECL_NAME)
+
+#INSTALL_ROOT = $(CURDIR)/debian/snuffleupagus
```

Comment:
Please don't commit a batch of comments.

Reply:
Commit pushed for that